main.tf

```
@@ -8,6 +8,26 @@ resource "azurerm_container_registry" "this" {
   tags = var.tags
 }
 
+resource "azurerm_container_registry_webhook" "this" {
+  for_each = { for object in var.webhook : object.name => object }
+
+  depends_on = [
+    azurerm_container_registry.this
+  ]
+
+  name                = each.value.name
+  registry_name       = var.registry_name
```

Get registry name from the container registry resource:

```suggestion
  name                = each.value.name
  registry_name       = azurerm_container_registry.this.name
```

This way, Terraform will create an automatic dependency between the webhook resources and the registry resource, removing the need of using the `depends_on` argument.